mozilla_django_oidc/auth.py

```diff
@@ -72,29 +72,60 @@ def create_user(self, claims):
 
         return self.UserModel.objects.create_user(username, email)
 
+    def get_algorithms(self):
+        """Returns the JWS verify algorithms"""
+        return ['HS256']
+
+    def get_verify_key(self):
+        """Returns the JWS verify key"""
+        return self.OIDC_RP_CLIENT_SECRET
+
     def verify_token(self, token, **kwargs):
         """Validate the token signature."""
+
         nonce = kwargs.get('nonce')
+        key = self.get_verify_key()
+        algorithms = self.get_algorithms()
 
         # Verify the token
-        verified_token = jws.verify(
-            token,
-            self.OIDC_RP_CLIENT_SECRET,
-            algorithms=['HS256']
-        )
+        verified_token = jws.verify(token, key, algorithms=algorithms)
         # The 'verified_token' will always be a byte string since it's
         # the result of base64.urlsafe_b64decode().
         # The payload is always the result of base64.urlsafe_b64decode().
         # In Python 3 and 2, that's always a byte string.
         # In Python3.6, the json.loads() function can accept a byte string
         # as it will automagically decode it to a unicode string before
         # deserializing https://bugs.python.org/issue17909
-        token_nonce = json.loads(verified_token.decode('utf-8')).get('nonce')
+        verified_id = json.loads(verified_token.decode('utf-8'))
+        token_nonce = verified_id.get('nonce')
 
-        if import_from_settings('OIDC_USE_NONCE', True) and nonce != token_nonce:
+        if import_from_settings('OIDC_USE_NONCE', True) \
+           and nonce != token_nonce:
             msg = 'JWT Nonce verification failed.'
             raise SuspiciousOperation(msg)
-        return True
+        return verified_id
+
+    def get_token_response(self, payload):
+        """Obtains the OIDC token"""
```

Comment:
/s/Obtains/Obtain/g

Reply:
Even better, something like:
```
"""Return token object as a dictionary."""
```
because it's not a "response" which is usually the word we use for HTTP things.

Reply:
done